pkgs/tools/misc/direnv/default.nix

```
@@ -16,15 +15,25 @@ buildGoModule rec {
   # we have no bash at the moment for windows
   BASH_PATH =
     lib.optionalString (!stdenv.hostPlatform.isWindows)
-    "${bash}/bin/bash";
+      "${bash}/bin/bash";
 
   # replace the build phase to use the GNUMakefile instead
   buildPhase = ''
     make BASH_PATH=$BASH_PATH
   '';
 
+  outputs = [ "out" ] ++ map (sh: "${sh}_interactiveShellInit");
+  shells = [ "bash" "fish" "zsh" ];
+
   installPhase = ''
-    make install PREFIX=$out
+        make install PREFIX=$out
+        ${lib.concatMapStrings
+    (
+        sh: ''
+          $out/bin/direnv hook ${sh} > $${sh}_interactiveShellInit
```

`$${foo}` escapes the `${foo}`.

```suggestion
          $out/bin/direnv hook ${sh} > ''$${sh}_interactiveShellInit
```